### PR TITLE
#524: Added the ability to specify the charset if needed

### PR DESCRIPTION
**Pull request by essobedo**

### CODE REVIEW

**Comment by davvd** [comment]
@essobedo Thanks, I will find someone to review this PR soon

**Comment by davvd** [comment]
@mkordas please review, thanks

**Comment by mkordas** [comment]
@essobedo I'm on it

**Comment by mkordas** [comment]
@essobedo first of all, please provide PR description

File: src/main/java/org/takes/rs/RsWithType.java [hunk 1]
```diff
@@ -58,6 +82,17 @@ public RsWithType(final Response res, final CharSequence type) {
      * Ctor.
      * @param res Original response
      * @param type Content type
+     *  @param charset The character set to add in the content type header
```
Comment (mkordas) [inline src/main/java/org/takes/rs/RsWithType.java]: @essobedo please align parameter

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 2]
```diff
@@ -222,5 +300,70 @@ public void replacesTypeWithText() throws Exception {
...
+                    ""
+                )
+            )
+        );
+    }
+
+    /**
+     * Ensures that the charset set is properly added when explicitly
+     * specified and absent when not specified.
+     * @throws java.lang.Exception If a problem occurs.
```
Comment (mkordas) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @essobedo `java.lang.` is not needed

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 3]
```diff
@@ -222,5 +300,70 @@ public void replacesTypeWithText() throws Exception {
...
+                    ),
+                    "",
+                    ""
+                )
+            )
+        );
+    }
+
+    /**
+     * Ensures that the charset set is properly added when explicitly
```
Comment (mkordas) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @essobedo please fix according to http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 4]
```diff
@@ -222,5 +300,70 @@ public void replacesTypeWithText() throws Exception {
...
+     * specified and absent when not specified.
+     * @throws java.lang.Exception If a problem occurs.
+     */
+    @Test
+    public void withCharset() throws Exception {
```
Comment (mkordas) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @essobedo we need verb here

File: src/test/java/org/takes/rs/RsWithTypeTest.java [hunk 5]
```diff
@@ -222,5 +300,70 @@ public void replacesTypeWithText() throws Exception {
...
+            Matchers.equalTo(
+                Joiner.on(CRLF).join(
+                    HTTP_OK,
+                    String.format(
+                        CONTENT_TYPE_WITH_CHARSET,
+                        TYPE_TEXT,
+                        StandardCharsets.ISO_8859_1
+                    ),
+                    "",
+                    ""
```
Comment (mkordas) [inline src/test/java/org/takes/rs/RsWithTypeTest.java]: @essobedo please put as much as possible to one line: http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#indentation

**Comment by mkordas** [comment]
@essobedo please to not close PRs by yourself, code review should be finished first

**Comment by essobedo** [comment]
@mkordas I closed this PR for the same reason I had to close https://github.com/yegor256/takes/pull/564

**Comment by mkordas** [comment]
@essobedo this causes too much noise, please never remove forks unless you are 100% confident that it cannot be solved differently

**Comment by essobedo** [comment]
@mkordas I agree with you, it was a mistake, it won't happen again. Please check my new PR which is https://github.com/yegor256/takes/pull/571

**Comment by mkordas** [comment]
@essobedo please at least mention in description issue for which this PR was raised

**Comment by davvd** [comment]
@ypshenychka please, review this ticket for compliance with our [QA rules](http://at.teamed.io/qa.html)

**Comment by essobedo** [comment]
@davvd @ypshenychka Please ignore this PR, it was a mistake on my side, the new PR for this issue is https://github.com/yegor256/takes/pull/571. Sorry for the noise.

**Comment by ypshenychka** [comment]
@essobedo 
According to our [QA Rules](http://at.teamed.io/qa.html):
>Pull request description explains the solution proposed and contains a link to the original ticket it is related to.

Please provide appropriate PR description.

**Comment by essobedo** [comment]
@ypshenychka Please ignore this PR, it was a mistake on my side, the new PR for this issue is https://github.com/yegor256/takes/pull/571. Sorry for the noise.

**Comment by mkordas** [comment]
@essobedo you need to update description anyway

**Comment by ypshenychka** [comment]
@essobedo I can't ignore the PR. @mkordas did his job and should be paid for it after I finish review.
So please do appropriate corrections to description.

**Comment by essobedo** [comment]
@ypshenychka As you wish, here is the description

**Comment by ypshenychka** [comment]
@essobedo Thanks

**Comment by ypshenychka** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@ypshenychka thanks for the review, we'll try better next time

**Comment by davvd** [comment]
@mkordas added 10 mins to @ypshenychka, for QA review, payment code is `76670654`; Thanks a lot, I just topped your account for **26 mins**, transaction ID `56b88a8e67df2a63880000fd`, total time was 15 hours and 26 mins; you're getting extra minutes here (c=11); +26 added to your rating, at the moment it is: [+5567](http://www.netbout.com/b/36080?open=rating)

**Pull request closed without merging**